src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace NServiceBus.Transports.RabbitMQ.Connection
+{
+    using System;
+    using global::RabbitMQ.Client;
+    using NServiceBus.Transports.RabbitMQ.Config;
+
+    class RabbitMqConnectionFactory
+    {
+        public ConnectionConfiguration Configuration { get; private set; }
+
+        public RabbitMqConnectionFactory(ConnectionConfiguration connectionConfiguration)
+        {
+            if (connectionConfiguration == null)
+            {
+                throw new ArgumentNullException("connectionConfiguration");
```

Comment:
OK, I'll leave it for now. I think I've come to a point with this PR that I'm polishing turds which may be removed entirely by a deeper refactoring.

It's sooo tempting to jump down the refactoring rabbit hole in this repo... :stuck_out_tongue: